tests/WE2E/utils.py

```diff
@@ -0,0 +1,584 @@
+                """))
+    else:
+        logging.fatal("Some kind of horrible thing has happened")
+        raise ValueError(dedent(
+              f"""Some kind of horrible thing has happened to the experiment status
+              for experiment {name}
+              status is {expt["status"]}
+              all task statuses are {statuses}"""))
+
+    # Final check for experiments where all tasks are "SUCCEEDED"; since the rocoto database does
+    # not include info on jobs that have not been submitted yet, use rocotostat to check that
+    # there are no un-submitted jobs remaining.
+    if expt["status"] in ["SUCCEEDED","STALLED","STUCK"]:
+        expt = compare_rocotostat(expt,name)
+
+    return expt
+
+def update_expt_status_parallel(expts_dict: dict, procs: int, refresh: bool = False,
+                                debug: bool = False) -> dict:
+    """
+    This function updates an entire set of experiments in parallel, drastically speeding up
+    the process if given enough parallel processes. Given an experiment dictionary, it will
+    output the updated dictionary.
+
+    parallelizes the call to update_expt_status across the given number of processes.
+    Making use of the python multiprocessing starmap functionality, takes
+
+    Args:
+        expts_dict (dict): A dictionary containing information for all experiments
+        procs      (int): The number of parallel processes
+        refresh   (bool): "Refresh" flag to pass to update_expt_status()
+        debug     (bool): Will capture all output from rocotorun. This will allow information such
+                          as job cards and job submit messages to appear in the log files, but can
+                          slow down the process drastically.
+
+    Returns:
+        dict: The updated dictionary of experiment dictionaries
+    """
+
+    args = []
+    # Define a tuple of arguments to pass to starmap
+    for expt in expts_dict:
+        args.append( (expts_dict[expt],expt,refresh,debug) )
+
+    # call update_expt_status() in parallel
+    with Pool(processes=procs) as pool:
+        output = pool.starmap(update_expt_status, args)
+
+    # Update dictionary with output from all calls to update_expt_status()
+    i = 0
+    for expt in expts_dict:
+        expts_dict[expt] = output[i]
+        i += 1
+
+    return expts_dict
+
+
+
+def print_test_info(txtfile: str = "WE2E_test_info.txt") -> None:
+    """Prints a pipe ( | ) delimited text file containing summaries of each test defined by a
+    config file in test_configs/*
+
+    Args:
+        txtfile (str): File name for test details file
+    """
+
+    testfiles = glob.glob('test_configs/**/config*.yaml', recursive=True)
+    testdict = dict()
+    links = dict()
+    for testfile in testfiles:
+        # Calculate relative cost of test based on config settings using legacy script
+        cost_array = calculate_cost(testfile)
```

Comment:
This was a product of legacy code that I did not feel was in the scope of this PR to address. Ideally `calculate_cost` should be completely re-written (I already had to make significant changes to get it working again) but this would require a lot of effort that I feel is best placed elsewhere